src/ui/GenioWindow.cpp

```diff
@@ -2357,15 +2391,25 @@ GenioWindow::_HandleNodeMonitorMsg(BMessage* msg)
 					if (path.Append(name.String()) == B_OK) {
 						entry.SetTo(path.Path());
 						if (entry.Exists()) {
-						// this will automatically send a file update.
+							// a B_STAT_CHANGED will follow.
 							return;
 						}
 					}
-
 				}
 			}
 
-			_HandleExternalRemoveModification(_GetEditorIndex(&nref));
+			// It could happens that the above check fails even if the file is there
```

Comment:
why not arming a timer. instead?

Reply:
Do you mean sending the 'second try' with a fixed delay?

Reply:
I was thinking of sending just one message after a given timeout. I have not understood why we need to check for a second try instead of just doint it once and then waiting for a timeout to expire.

Reply:
It's done twice to avoid delays  if not necessary. We receive the event and maybe we are already ok and we can avoid rechecking it later. If we always send delayed message we will always wait.